app/src/main/java/com/github/se/cyrcle/ui/report/ParkingReportScreen.kt

```diff
@@ -0,0 +1,188 @@
+package com.github.se.cyrcle.ui.report
+
+import android.annotation.SuppressLint
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.parking.ParkingReport
+import com.github.se.cyrcle.model.parking.ParkingReportReason
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.user.UserViewModel
+import com.github.se.cyrcle.ui.addParking.attributes.DESCRIPTION_MAX_LENGTH
+import com.github.se.cyrcle.ui.addParking.attributes.DESCRIPTION_MIN_LENGTH
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.theme.Typography
+import com.github.se.cyrcle.ui.theme.atoms.ConditionCheckingInputText
+import com.github.se.cyrcle.ui.theme.atoms.Text
+import com.github.se.cyrcle.ui.theme.disabledColor
+import com.github.se.cyrcle.ui.theme.molecules.EnumDropDown
+import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ParkingReportScreen(
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel,
+    parkingViewModel: ParkingViewModel,
+) {
+  val configuration = LocalConfiguration.current
+  val screenWidth = configuration.screenWidthDp.dp
+  val screenHeight = configuration.screenHeightDp.dp
+  var showDialog by remember { mutableStateOf(false) }
+  val context = LocalContext.current
+
+  // Define padding as a percentage of screen dimensions
+  val horizontalPaddingScaleFactor = screenWidth * 0.03f
+  val topBoxHeight = screenHeight * 0.10f // 10% of screen height for top box
+  val verticalPaddingScaleFactor = screenHeight * 0.02f
+
+  // State for report inputs
+  val selectedReason = rememberSaveable { mutableStateOf(ParkingReportReason.INEXISTANT) }
+  val parkingId = parkingViewModel.selectedParking.value?.uid
+  val reportDescription = rememberSaveable { mutableStateOf("") }
+  val userId = userViewModel.currentUser.value?.public?.userId
+
+  fun onSubmit() {
+    val report =
+        ParkingReport(
+            uid = parkingViewModel.getNewUid(),
+            reason = selectedReason.value,
+            userId = userId ?: "",
```

Comment:
It's weird to let the user id field empty if no user id was found. Maybe is it better to enforce the fact that a userID must be linked to a report. What do you think ?